### ISSUE TITLE
Missing doctrine migrations

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#1374]
### Description

The latest Neos base distribution misses some doctrine migrations

### Steps to Reproduce

1. [Download](https://discuss.neos.io/t/development-setup/504) latest Neos base distribution
2. Walk through setup
3. Run `./flow doctrine:migrationgenerate`

#### Expected behavior

The generated migration should be empty because all provided migrations are executed during setup

#### Actual behavior

A 300-line migration is created, mostly dropping and re-adding indexes like:
https://gist.github.com/bwaidelich/ff23b9fd0be3cc38cf3265c916cf431c

### Affected Versions

Neos master

**Comment by skurfuerst** [comment]
OK... so what happens here? maybe due to a doctrine update or so?

**Comment by bwaidelich** [comment]
At first I thought it just changed the casing (removing lower case, re-adding upper case) but it's in fact the hash that changes:
```
DROP INDEX idx_60a956b98d940019 ON neos_contentrepository_domain_model_nodedata
CREATE INDEX IDX_CE6515698D940019 ON neos_contentrepository_domain_model_nodedata (workspace)
```

**Comment by bwaidelich** [comment]
OK, it seems the diff comes from `Doctrine\DBAL\Schema\AbstractAsset::_generateIdentifierName()`, but it's not a change in behavior but the result of the renamed tables:
```php
protected function _generateIdentifierName($columnNames, $prefix='', $maxSize=30)
{
	$hash = implode("", array_map(function ($column) {
		return dechex(crc32($column));
	}, $columnNames));

	return substr(strtoupper($prefix . "_" . $hash), 0, $maxSize);
}
```

The method gets called with `['neos_contentrepository_domain_model_nodedata', 'movedto']` for example. Previously the first array element was `typo3_typo3cr_domain_model_nodedata` obviously thus the change.
But I still wonder:

1. Why wasn't this part of the "package rename doctrine migrations"?
2. Why is the table name passed as part of `$columnNames` above, is that as intended?

**Comment by bwaidelich** [comment]
Re:
> 2. Why is the table name passed as part of $columnNames above, is that as intended?

Yes, that is how it's implemented in Doctrine and I assume it's required to avoid ambiguity .
But now

3. Is it our only option to add migrations re-adding the indexes (which can take a very long time on large dbs)?

**Comment by bwaidelich** [comment]
Re
> 3. Is it our only option to add migrations re-adding the indexes (which can take a very long time on large dbs)?

from @beheist via Slack:

> rename index is possible in mysql as well, but only on 5.7+ (as ALTER TABLE xyz RENAME INDEX…) (edited)

> which can be nasty when you create migrations on 5.7 and try to run them on an earlier version, we had that in a project a while ago

> But that could improve performance of the necessary migrations at least for those using 5.7+

**Comment by skurfuerst** [comment]
Hey,

so I guess we need to:

- adjust our core packages, to use "RENAME INDEX" whereever possible.
- provide instructions to generate a migration after the upgrade is complete.

All the best,
Sebastian

**Comment by bwaidelich** [comment]
> adjust our core packages, to use "RENAME INDEX" whereever possible.

yes, I push a follow-up to the 3 PRs using

```php
if ($this->connection->getDatabasePlatform() instanceof MySQL57Platform) {
// …
} else {
// …
}
```

> provide instructions to generate a migration after the upgrade is complete.

What do you mean by that?

**Comment by bwaidelich** [comment]
OK, I added the platform check to all migrations of the PRs and succesfully tested it with MySQL 5.1.72, MySQL 5.7.17 & Postgresql 9.5.3

**Comment by skurfuerst** [comment]
ah you are right. we do not change namespaces of foreign code. Fine :D <3

**Comment by bwaidelich** [comment]
Resolved with #1375, neos/flow-development-collection#819 and neos/party#15
